server/app/lib/utils/fusor/discover_macs.rb

```diff
@@ -0,0 +1,55 @@
+require 'net/ssh'
+require 'nokogiri'
+require 'rubyipmi'
+
+module Utils
+  module Fusor
+    class DiscoverMacs
+      def initialize(params)
+        @params = params
+      end
+
+      def discover
+
+        if @params[:driver] == 'pxe_ssh'
+          kvm(@params)
+        elsif @params[:driver] == 'pxe_ipmitool' && @params[:vendor] == 'dell'
+          dell(@params)
+        end
+
+      end
+
+      private
+
+      def kvm(params)
+        begin
+          client = Net::SSH.start(params[:hostname], params[:username], :password => params[:password])
+          libvirt_domains = client.exec!('virsh list --all --name').split("\n")
```

Comment:
Any reason not to reuse Fusor::SSHConnection?  https://github.com/fusor/fusor/blob/master/server/app/lib/utils/fusor/ssh_connection.rb